larq/optimizers_v2.py

```diff
@@ -40,44 +119,21 @@ class Bop(tf.keras.optimizers.Optimizer):
     - [Latent Weights Do Not Exist: Rethinking Binarized Neural Network Optimization](https://arxiv.org/abs/1906.02107)
     """
 
-    def __init__(self, fp_optimizer, threshold=1e-7, gamma=1e-2, name="Bop", **kwargs):
+    def __init__(self, threshold=1e-7, gamma=1e-2, name="Bop", **kwargs):
         super().__init__(name=name, **kwargs)
 
-        if not isinstance(fp_optimizer, tf.keras.optimizers.Optimizer):
-            raise TypeError(
-                f"Expected tf.keras.optimizers.Optimizer, received {type(fp_optimizer)}."
-            )
-
-        self.fp_optimizer = fp_optimizer
         self._set_hyper("threshold", threshold)
         self._set_hyper("gamma", gamma)
 
+    # TODO: Why do we need to do this here? (It gets called in apply_gradients().)
+    # TODO: Can we remove the `is_binary` check? Will this never get called with non-
+    # binary variables anyway?
     def _create_slots(self, var_list):
         for var in var_list:
-            if self.is_binary(var):
+            if OptimizerGroup.is_binary(var):
```

Comment:
`apply_gradients` only gets passed the binary vars, so there should be no additional check needed here

Reply:
Check removed in 51a20ab.